src/redux/missions/missionsSlice.js

```diff
@@ -0,0 +1,57 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  missions: [],
+  isLoading: true,
+  error: null,
+};
+
+const url = 'https://api.spacexdata.com/v3/missions';
+
+export const getMissions = createAsyncThunk('missions/getMissions', async () => {
+  try {
+    const resp = await axios.get(url);
+    return resp.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+const missionsSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    MemberShip: (state, action) => {
+      const newMission = state.missions.map((mission) => {
+        if (mission.mission_id !== action.payload) { return mission; }
+        if (mission.reserved === true) {
+          return { ...mission, reserved: false };
+        }
+        return { ...mission, reserved: true };
+      });
+      state.missions = newMission;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(getMissions.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(getMissions.fulfilled, (state, action) => {
+        const newState = action.payload.map((mission) => ({ ...mission, reserved: false }));
+        return ({
+          ...state,
+          missions: newState,
+          isLoading: false,
+        });
+      })
```

Comment:
- Kindly, you are storing the whole API response object which is not necessary, Please filter and store only used data to redux store to improve its performance 👌 
![2023-04-15_04-47_1](https://user-images.githubusercontent.com/52010789/232177352-f87d198b-8e13-4895-9997-980a643b36b6.png)